instrumentation/messaging/src/main/java/brave/messaging/MessagingTracing.java

```diff
@@ -126,13 +141,23 @@ public Builder producerSampler(SamplerFunction<MessagingRequest> producerSampler
       return this;
     }
 
-    /** @see MessagingTracing#consumerSampler() */
+    /**
+     * @see MessagingTracing#consumerSampler()
+     */
     public Builder consumerSampler(SamplerFunction<MessagingRequest> consumerSampler) {
       if (consumerSampler == null) throw new NullPointerException("consumerSampler == null");
       this.consumerSampler = consumerSampler;
       return this;
     }
 
+    /**
+     * @see MessagingTracing#newTraceOnReceive()
+     */
+    public Builder newTraceOnReceive(boolean newTraceOnReceive) {
```

Comment:
I think this might need to be an expression, similar to a sampling function?

Reply:
Yes! sampler function is a very similar interface. We can add a similar design here.

Would make sense to discuss something as https://cloud.google.com/trace/docs/reference/v2/rest/v2/Links supporting this links as a Zipkin/Brave type? Or tags would be enough (e.g. `parent.trace.id=... parent.span.id=...`)?

Reply:
I think the links thing is being inherited into otel. I haven't followed that part, but in opencensus it was hardly defined. Ex not sure it is for this case or something else. Would appreciate if you could summarize later on otel, but for now we can chat about stackdriver.

At any rate, we have some choices in Brave as we don't have to map 1-1 with zipkin model. For example, `MutableSpan.error()` In other words, let's get the api right first then consider normalizing a model.

I suspect if the semantics match what links are supposed to be used for, we can consider updating zipkin-gcp accordingly ( I think at some point that code should implement a SpanHandler anyway). Ex, the direction is important. I think in the stackdriver they discuss a batchhandler who is in an existing trace, and notices inbound trace IDs from messages. This is not the same as what I believe we are discussing here. Here we only have one inbound and one outbound trace, We are choosing to sever it in order to limit the tree (this use case needs to be fleshed out actually). Maybe you can add a RATIONALE file with all the use cases you've seen?

@saturnism do you know if the intent of links are for the case where you intentionally break a trace from an inbound message in order for "processor down" to not cause the upstream trace to be massive?